docs/rules/no-unnecessary-polyfills.md

```diff
@@ -0,0 +1,76 @@
+# Enforce the use of built-in methods instead of unnecessary polyfills
+
+<!-- Do not manually modify RULE_NOTICE part. Run: `npm run generate-rule-notices` -->
+<!-- RULE_NOTICE -->
+✅ *This rule is part of the [recommended](https://github.com/sindresorhus/eslint-plugin-unicorn#recommended-config) config.*
+<!-- /RULE_NOTICE -->
+
+This rules helps to use existing methods instead of using extra polyfills.
+
+## Fail
+
+package.json
+
+```json
+{
+	"engines": {
+		"node": ">=8"
+	}
+}
+```
+
+```js
+const assign = require('object-assign');
+```
+
+## Pass
+
+package.json
+
+```json
+{
+	"engines": {
+		"node": "4"
+	}
+}
+```
+
+```js
+const assign = require('object-assign'); // passes as Object.assign is not supported
+```
+
+## Options
+
+Type: `object`
+
+### targets
+
+Type: `string | string[] | object`
+
+The `targets` option allows to specify the target versions. This option could be a Browserlist query or a targets object, see [core-js-compat `targets` option](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js-compat#targets-option) for more informations. It could also be a Node.js target version (SemVer syntax supported) if the [`treatsTargetsAsSemver` option](#treatsTargetsAsSemver) is set to `true`.
```

Comment:
I think we should only allow `targets`, for `engines.node` field, we can turn it into a browser target. So we don't need `treatsTargetsAsSemver`.

Reply:
Do you mean to only allow using semver syntax in `targets.node` and remove `treatsTargetsAsSemver`?

```js
"unicorn/no-unnecessary-polyfills": ["error", {	"targets": { "node": "..semver.." } }] // valid
"unicorn/no-unnecessary-polyfills": ["error", {	"targets": "..semver.." }] // invalid
```

Reply:
No, I mean only [core-js "target"](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js-compat#targets-option) , for `engine.node` in package.json, we only need translate to it (maybe?). So we don't need `treatsTargetsAsSemver`

Reply:
I'm sorry, I'm having trouble understanding, do you have any examples please?

Reply:
Sorry I didn't read carefully, yes your example is correct


```
"unicorn/no-unnecessary-polyfills": ["error", {	"targets": { "node": "..semver.." } }] // valid
"unicorn/no-unnecessary-polyfills": ["error", {	"targets": "..semver.." }] // invalid
```

Reply:
We don't  need care what targets is, use it directly.

Reply:
Thanks, it's clearer. PTAL